sdk/language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -138,27 +145,35 @@ main = do
           Left err -> fail . T.unpack $ err
           Right pkgMap -> do
               forM_ (Map.toList pkgMap) $
-                \(pkgId, (mbPkgName, pkg)) -> do
-                     let id = unPackageId pkgId
-                         pkgName = packageNameText pkgId mbPkgName
-                     let pkgDesc = case mbPkgName of
-                           Nothing -> id
-                           Just pkgName -> unPackageName pkgName <> " (hash: " <> id <> ")"
+                \(pkgId, (pkgRef, pkg)) -> do
+                     let pkgDesc = case pkgRef of
+                           PkgId _ -> unPackageId pkgId
+                           PkgNameVer (pkgName, _) -> unPackageName pkgName <> " (hash: " <> unPackageId pkgId <> ")"
                      T.putStrLn $ "Generating " <> pkgDesc
                      daml2js Daml2jsParams{..}
 
-packageNameText :: PackageId -> Maybe PackageName -> T.Text
```

Comment:
`packageNameText` is no longer sufficient, as we now have cases where we want just the name part (without the version) for a template id. This is why we create and pass around this new `PackageReference` type, and use the appropriate function, i.e. either `forFilePath` or `forTemplateId` to convert it into text for the appropriate purpose.